### DISCUSSION TITLE
How to render React Countdown Circle Timer with useCountdown?

### DISCUSSION THREAD

**Discussion by DanielGennat** [neuefische/questions#177] [Web] [answered]
Hi, I have no idea how to render the timer with useCountdown. This ist the [documentation](https://www.npmjs.com/package/react-native-countdown-circle-timer).
Thanks in advance :)

**Comment by MerleMak** [comment]
Hi Daniel, seems like this package is optimised for React Native. Maybe you can find something similar for NextJs?  :) 
What is the goal you are trying to achieve by implementing the useCountdown?

**Reply by DanielGennat** [reply]
Hi Merle,
I think [this](https://www.geeksforgeeks.org/how-to-add-timer-in-next-js/) was the documentation for nextJs I followed installing the timer component.

`onComplete` I want to start the timer with a new interval.
`onComplete={() => ({ shouldRepeat: newInitialRemainingTime: 5})}` works but it won't adapt the animation of the timer.
So I think I have to use `setKey((prevKey) => prevKey + 1);` to render a new timer and use `useCounter` to change the duration...

Here is a [pull request](https://github.com/DanielGennat/capstone-project/pull/11)

**Reply by DanielGennat** [reply]
I try to write my own useStates now

**Reply by DanielGennat** [reply]
Here is my try and it works:

key and onComplete in the timer component:
`key={key}`
`duration={duration}`
`onComplete={setUpNextInterval}`

array, useStates and handle function inside the Home function:

```
const timerChain = [3, 1, 2, 1];
  const [timerPointer, setTimerPointer] = useState(1);

  const [key, setKey] = useState(0);
  const [duration, setDuration] = useState(timerChain[0]);
```

```
function setUpNextInterval() {
    if (timerPointer === timerChain.length) {
    } else {
      setKey((prevKey) => prevKey + 1);
      setTimerPointer(timerPointer + 1);
      setDuration(timerChain[timerPointer]);
    }
  }
```

**Comment by MerleMak** [comment, marked as answer]
Hi Daniel,

So you can actually fix this without useCounter. 

**Concerning the animation:**
First of all there are some things to consider with your CountdownCircleTimer Component:
the duration is fixed at 3. Apparently the prop `duration` is determining the time it takes for the animation to complete the circle. 

In order to be able to display the right animation for the respective Timer, you first of all need to work with a place to store the different timers, this can be an **array** to make it work for now. 

So
1. Inside of your Home function you will need to define an array, with some example timers. I worked with an array of 5 different `timers` with random times, like 4, 5, 3, 1, 2 (seconds).

2. To be able to access the different example timers you also need a `useState`, e.g. currentTimer with an initial value of **0**. This State you will use to write logic to set up the next interval but also for the `duration` and `key` prop in your Countdown Component.

3. The key prop needs to access the state of {currentTimer}, so have a unique key for each timer to work with. The duration prop need to access the currentTimer within the array of timers ({timers[currentTimer]}.

**Concerning the logic to set up the next interval:**
Now we already have an example array and a new useState. I think you were already on the right way with setKey and prevKey, but i think there is a simpler way. 

Inside of the function `setUpNextInterval` you want a logic that checks if the index of the current Timer (which is saved in the currentTimer UseState) is still inside of the array `timers`. When this is the case you want to set the index of CurrentTimer to +1
Its easiest to check this within a condition

That should do the trick :)
Let me know if anything is unclear!

**Reply by DanielGennat** [reply]
Hi Merle,
thank you for the detailed answer!
You are right! `key` and `duration` states are not necessary so i can reduce the logic.

key and onComplete in the timer component:
`key={timerPointer}`
`duration={timerChain[timerPointer]}`
`onComplete={setUpNextInterval}`

array, useStates and handle function inside the Home function:

```
  const timerChain = [3, 1, 2, 1];
  const [timerPointer, setTimerPointer] = useState(0);
```
```
function setUpNextInterval() {
    if (timerPointer + 1 === timerChain.length) {
    } else {
      setTimerPointer(timerPointer + 1);
    }
  }
```
